wire-ios/Wire-iOS/Sources/UserInterface/Conversation/Content/Cells/Utility/MessageToolboxDataSource.swift

```diff
@@ -114,14 +112,18 @@ final class MessageToolboxDataSource {
         }
         // 2) Failed to send
         else if failedToSend && isSentBySelfUser {
-            var detailsString: String
+            typealias Message = L10n.Localizable.Content.System.FailedtosendMessage
 
+            let detailsString: String
             switch message.failedToSendReason {
             case .unknown, .none:
-                detailsString = FailedToSendMessage.generalReason
+                let userCancelled = message.fileMessageData?.transferState == .uploadingCancelled
```

Comment:
I'm wondering if instead of inspecting the transfer state here, couldn't we do it at the time of failure and set an appropriate `failedToSendReason`, such as

```
.uploadCancelled
```

Reply:
That would certainly make more sense. For come reason I thought that `failedToSendReason` is synced across devices. Just to confirm this is only local and so I can add more cases (as long as I don't change there int values) right?

Reply:
I've changed implementation following your suggestion